File: c/space-age/space_age.c
#include "space_age.h"

#define ERROR_VALUE -1.0

static const float PLANET_YEAR_RATIOS[] = {
    0.2408467,  // Mercury year
    0.61519726, // Venus year
    1.0,        // Earth year
    1.8808158,  // Mars year
    11.862615,  // Jupiter year
    29.447498,  // Saturn year
    84.016846,  // Uranus year
    164.79132   // Neptune year
};

static const int SECONDS_A_YEAR = 31557600;

float age(planet_t planet, int64_t seconds)
{
    if (planet >= 0 && planet < 8) {
        return seconds / SECONDS_A_YEAR / PLANET_YEAR_RATIOS[planet];
    } else {
        return ERROR_VALUE;
    }
}

File: go/space-age/space_age.go
package space

type Planet string

const secondsInAYear = 31557600

var yearOn = map[Planet]float64{
    "Mercury": 0.2408467,
    "Venus":   0.61519726,
    "Earth":   1.0,
    "Mars":    1.8808158,
    "Jupiter": 11.862615,
    "Saturn":  29.447498,
    "Uranus":  84.016846,
    "Neptune": 164.79132,
}

func Age(seconds float64, planet Planet) float64 {
    ratio, ok := yearOn[planet]
    if !ok {
        return -1.0
    }

    return seconds / secondsInAYear / ratio
}
